src/components/TaxServiceModal/TaxServiceBanner.tsx

```diff
@@ -0,0 +1,165 @@
+import styled from 'styled-components/macro'
+import { Z_INDEX } from 'theme/zIndex'
+
+import TaxServiceModal from '.'
+import CointrackerLogo from './CointrackerLogo.png'
+import TokenTaxLogo from './TokenTaxLogo.png'
+
+const PopupContainer = styled.div<{ show: boolean; isDarkMode: boolean }>`
+  box-shadow: ${({ theme }) => theme.deepShadow};
+  border: 1px solid ${({ theme }) => theme.backgroundOutline};
+  background-color: ${({ theme }) => theme.backgroundSurface};
+  border-radius: 13px;
+  cursor: pointer;
+  color: ${({ theme }) => theme.textPrimary};
+  display: ${({ show }) => (show ? 'flex' : 'none')};
+  flex-direction: column;
+  position: fixed;
+  right: clamp(0px, 1vw, 16px);
+  z-index: ${Z_INDEX.sticky};
+  transition: ${({
+    theme: {
+      transition: { duration, timing },
+    },
+  }) => `${duration.slow} opacity ${timing.in}`};
+  width: 320px;
+  height: 156px;
+  @media screen and (min-width: ${SMALLEST_MOBILE_MEDIA_BREAKPOINT}) {
+    bottom: 50px;
+  }
+
+  ::before {
+    content: '';
+    position: absolute;
+    top: 0px;
+    right: 0px;
+    bottom: 0px;
+    left: 0px;
+
+    background-image: url(${CointrackerLogo}), url(${TokenTaxLogo});
+    background-size: 15%, 20%;
+    background-repeat: no-repeat;
+    background-position: top right 75px, bottom 5px right 7px;
+
+    opacity: ${({ isDarkMode }) => (isDarkMode ? '0.9' : '0.25')};
+  }
+`
+
+const InnerContainer = styled.div`
+  border-radius: 12px;
+  cursor: auto;
+  overflow: hidden;
+  display: flex;
+  flex-direction: column;
+  position: relative;
+  gap: 8px;
+  padding: 16px;
+  ::before {
+    content: '';
+    position: absolute;
+    top: 0px;
+    right: 0px;
+    bottom: 0px;
+    left: 0px;
+
+    background-color: ${({ theme }) => theme.accentActionSoft};
+    opacity: 0.4;
+  }
+`
+
+const Button = styled(ThemeButton)`
+  margin-top: auto;
+  margin-right: auto;
+  padding: 8px 24px;
+  gap: 8px;
+  border-radius: 12px;
+`
+
+const TextContainer = styled.div`
+  user-select: none;
+  display: flex;
+  flex-direction: column;
+  width: 70%;
+  justify-content: center;
+`
+
+export const StyledXButton = styled(X)`
+  color: ${({ theme }) => theme.textPrimary};
+  cursor: pointer;
+  &:hover {
+    opacity: ${({ theme }) => theme.opacity.hover};
+  }
+  &:active {
+    opacity: ${({ theme }) => theme.opacity.click};
+  }
+`
+
+const TAX_SERVICE_DISMISSED = 'TaxServiceToast-dismissed'
+
+export default function TaxServiceBanner() {
+  const isDarkMode = useIsDarkMode()
+  const [modalOpen, setModalOpen] = useState(false)
+  const sessionStorageTaxServiceDismissed = sessionStorage.getItem(TAX_SERVICE_DISMISSED)
+  let initialBannerOpen = true
+  if (!sessionStorageTaxServiceDismissed) {
+    sessionStorage.setItem(TAX_SERVICE_DISMISSED, 'false')
+  } else if (sessionStorageTaxServiceDismissed === 'true') {
+    initialBannerOpen = false
+  }
+  const [bannerOpen, setBannerOpen] = useState(initialBannerOpen)
```

Comment:
```suggestion

  if (!sessionStorageTaxServiceDismissed) {
    sessionStorage.setItem(TAX_SERVICE_DISMISSED, 'false')
  }
  const [bannerOpen, setBannerOpen] = useState(sessionStorageTaxServiceDismissed === 'true')
```

Reply:
looks good, I changed last line to `const [bannerOpen, setBannerOpen] = useState(sessionStorageTaxServiceDismissed !== 'true')` but the gist makes sense thanks